src/compiler/types.ts

```diff
@@ -128,6 +128,7 @@ module ts {
         PrivateKeyword,
         ProtectedKeyword,
         PublicKeyword,
+        AbstractKeyword,

```

Comment:
this is not a future reserved word in ES6, should move to the contextual keywords block. you should also add a test for variables named "abstract" with "use strict";